pkgs/tools/misc/diffoci/default.nix

```diff
@@ -16,12 +19,30 @@ buildGoModule rec {
 
   vendorHash = "sha256-18rsa91PiqZv70EK3K6K1l6N2mIpoVpkX29amKCo5cg=";
 
-  ldflags = [ "-s" "-w" ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=github.com/reproducible-containers/${pname}/cmd/${pname}/version.Version=v${version}"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall =
+    let
+      diffoci = if stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform then "\"$out\"" else buildPackages.diffoci;
+    in
+    ''
+      installShellCompletion --cmd ${pname} \
+        --bash <(${diffoci}/bin/${pname} completion bash) \
+        --fish <(${diffoci}/bin/${pname} completion fish) \
+        --zsh <(${diffoci}/bin/${pname} completion zsh)
+    '';
 
   meta = with lib; {
     description = "Diff for Docker and OCI container images";
     homepage = "https://github.com/reproducible-containers/diffoci/";
     license = licenses.asl20;
+    mainProgram = pname;
```

Comment:
Unfortunately, `meta.mainProgram` has already been added to `diffoci` as part of #270366 (which seemingly opened a month after your PR).

Meaning, this PR will no longer eval if you rebase it on the latest master, because `mainProgram` is defined twice (in two different lines).

Please rebase and then remove your `mainProgram` addition in favor of the one from #270366, so eval will continue to work.

(Yes, it's annoying, sorry)

Reply:
Done, no worries :smile: